--- a/extensions/renderer/isolated_world_manager.cc
+++ b/extensions/renderer/isolated_world_manager.cc
@@ -95,8 +95,6 @@ void IsolatedWorldManager::SetUserScript
       info.human_readable_name =
           blink::WebString::FromUTF8(isolated_world.name);
       info.stable_id = blink::WebString::FromUTF8(host_id);
-      info.content_security_policy =
-          blink::WebString::FromUTF8(*pending_info.csp);
       blink::SetIsolatedWorldInfo(blink_world_id, info);
     }
   }
@@ -233,10 +231,6 @@ void IsolatedWorldManager::UpdateBlinkIs
   blink_info.security_origin = blink::WebSecurityOrigin::Create(world_info.url);
   blink_info.human_readable_name = blink::WebString::FromUTF8(world_info.name);
   blink_info.stable_id = blink::WebString::FromUTF8(world_info.host_id);
-  if (world_info.csp) {
-    blink_info.content_security_policy =
-        blink::WebString::FromUTF8(*world_info.csp);
-  }
   blink::SetIsolatedWorldInfo(world_id, blink_info);
 }
 
